### PR TITLE
Update calico to 3.16.4

### PR DESCRIPTION
**Pull request by ipochi**
This PR updates calico to 3.16.4

### CODE REVIEW

File: assets/charts/control-plane/calico/templates/calico-kube-controllers.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# See https://github.com/projectcalico/kube-controllers
```
Comment (surajssd) [inline assets/charts/control-plane/calico/templates/calico-kube-controllers.yaml:1]: Was this change on purpose?

Comment (surajssd) [reply assets/charts/control-plane/calico/templates/calico-kube-controllers.yaml:1]: I mean moving from deployment.yaml to this file?

Comment (ipochi) [reply assets/charts/control-plane/calico/templates/calico-kube-controllers.yaml:1]: No I mean , this makes the file naming correct to the resource it holds.

**Review by surajssd — APPROVED** [review]
ok

File: assets/charts/control-plane/calico/templates/calico-kube-controllers.yaml (hunk 1 above)
Comment (ipochi) [reply assets/charts/control-plane/calico/templates/calico-kube-controllers.yaml:1]: the file was named `deployment.yaml`, it contain the deployment resource. I renamed it reflect better.
Similar with `daemonset.yaml`

deployment.yaml - calico-kube-controllers.yaml
daemonset.yaml - calico-node.yaml

File: assets/charts/control-plane/calico/crds/bgpconfigurations.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ kind: CustomResourceDefinition
 metadata:
```
Comment (invidian) [inline assets/charts/control-plane/calico/crds/bgpconfigurations.yaml:3]: nit:
```diff
- calico: update calico CRDS
+ calico: update calico CRDs
```

File: assets/charts/control-plane/calico/templates/cluster-role-binding.yaml [hunk 3]
```diff
@@ -1,3 +1,4 @@
+---
```
Comment (invidian) [inline assets/charts/control-plane/calico/templates/cluster-role-binding.yaml:1]: Honestly, the best would be to have everything in `calico.yaml`, as from `wget https://docs.projectcalico.org/v3.16.4/manifests/calico.yaml`

**Review by invidian — APPROVED** [review]
Just some nits, otherwise LGTM.

File: assets/charts/control-plane/calico/templates/cluster-role-binding.yaml (hunk 3 above)
Comment (ipochi) [reply assets/charts/control-plane/calico/templates/cluster-role-binding.yaml:1]: IMO, we should look into integrating Tigera Operator in our helm chart.

Comment (ipochi) [reply assets/charts/control-plane/calico/templates/cluster-role-binding.yaml:1]: why not ?

Comment (invidian) [reply assets/charts/control-plane/calico/templates/cluster-role-binding.yaml:1]: Hm, perhaps this is the way to go.
